### ISSUE TITLE
MassiveSearch Document title should also be indexed for search boost

### ISSUE THREAD

**Issue by ampaze** [sulu/sulu#5290]
| Q | A
| --- | ---
| Sulu Version | 1.6.33

# Actual Behavior

As far as I can see, it is impossible to influence the search result order by making the title or url more important.

To work around this I wanted to give the title a boost directly in the search query. (something like  `__title:("test")^2`)

Unfortunately this is also not possible right now, as the __title is only stored but not indexed.

I have the following property definition
```
<property name="headline" type="text_line" mandatory="true">
    <meta>
        <title lang="de">Überschrift</title>
    </meta>
    <params>
        <param name="headline" value="true"/>
    </params>
    <tag name="sulu.search.field" role="title" />
</property>
```

When I dump the result of a search (e.g. a `ZendSearch\Lucene\Document` object)

> "__title" => Field {
      +name: "__title"
      +value: "xxxxxxx"
      +isStored: true
      +isIndexed: 0
      +isTokenized: 0
      +isBinary: false
      +storeTermVector: false
      +boost: 1.0
      +encoding: "UTF-8"
    }

it shows, that the property is not indexed.

# Expected Behavior

The title search field should be indexed, same goes for the url in my opinion.
I want to be able to query for the title explicitly, which is not possible.

Alternatively having any kind of option of giving a search field a boost or a weight would also be a step in the right direction.

**Comment by almare** [comment]
This would be great! Definitely gives that more control over the content and better search results

**Comment by alexander-schranz** [comment]
@almare can you show the complete `.xml` you use?

A role can only exist once and search fields whith roles should only link to root properties so `title`, `description`, `image` should only exist once in your .xml. That also means role based search fields can not be inside a block.
If you just want to index a field you should just add the tag without a role:

```xml
<tag name="sulu.search.field" />
```

**Comment by almare** [comment]
Ah, sorry. My comment was not clear. My comment was  regarding the boost feature ^2 . Indeed I have a lot of content managed by blocks. Blocks are mainly my whole content, because they are so flexible and reusable. It would be nice if I could boost them somehow. That was my intention.


[homepage.xml.txt](https://github.com/sulu/sulu/files/4650794/homepage.xml.txt)

**Comment by alexander-schranz** [comment]
Oh sorry I misinterpreted not being indexed. Currently all data is indexed in a merged content field. But I think it would make sense to make title and description also being indexed and so able being boosted.

**Comment by almare** [comment]
Regarding the boost option. This is definitely a feature and gives more control.

**Comment by ampaze** [comment]
Something like this

`<tag name="sulu.search.field" weight="2"/>`

would be even better.

**Comment by almare** [comment]
Just my 2 cents 
`<tag name="sulu.search.field" boost="2"/>`
Would be more clear.

**Comment by ampaze** [comment]
I could try to implement this, if you tell me which files I should take a look at. It's not really straight forward to understand all the different parts that work together for this.

**Comment by danrot** [comment]
Fixed in #5304
